src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs

```diff
@@ -0,0 +1,911 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.ConvertNamespace;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertNamespace
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<ConvertNamespaceCodeRefactoringProvider>;
+
+    public class ConvertNamespaceRefactoringTests
+    {
+        #region Convert To File Scoped
+
+        [Fact]
+        public async Task TestNoConvertToFileScopedInCSharp9()
+        {
+            var code = @"
+namespace $$N
+{
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp9,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.BlockScoped }
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestNoConvertToFileScopedInCSharp10WithFileScopedPreference()
+        {
+            var code = @"
+namespace $$N
+{
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp10,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.FileScoped }
```

Comment:
Isn't this backwards? This looks like the preference is for file scoped, so why wouldn't we over conversion when the namespace is block scoped?

Reply:
This is the refactoring. It is offered to rename you to go in the direction *opposite* your preference. The analyzer/fixer push your code in the direction of your style choice.